addon/services/redux.js

```diff
@@ -0,0 +1,50 @@
+import Ember from 'ember';
+import redux from 'redux';
+import reducers from '../reducers/index';
+import enhancers from '../enhancers/index';
+import middlewareConfig from '../middleware/index';
+
+const { assert, isArray, get } = Ember;
+
+// Handle "classic" middleware exports (i.e. an array), as well as the hash option
+const extractMiddlewareConfig = (mc) => {
+  assert(
+    'Middleware must either be an array, or a hash containing a `middleware` property',
+    isArray(mc) || mc.middleware
+  );
+  return isArray(mc) ? { middleware: mc } : mc;
+}
+
+const { createStore, applyMiddleware, combineReducers, compose } = redux;
+
+const makeStoreInstance = ({middlewareConfig, reducers, enhancers}) => {
+  const { middleware, setup = () => {} } = extractMiddlewareConfig(middlewareConfig);
+  const createStoreWithMiddleware = compose(applyMiddleware(...middleware), enhancers)(createStore);
+  const reducer = typeof reducers === 'function' ? reducers : combineReducers(reducers);
```

Comment:
only because I'm 99% sure I'll merge that PR in the next 24 hours and I'll rebase this one right after (sorry for the confusion here - mostly trying to work conflict free)